community/kernel/src/main/java/org/neo4j/storageengine/api/schema/IndexProgressor.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.storageengine.api.schema;
+
+import org.neo4j.internal.kernel.api.LabelSet;
+import org.neo4j.values.storable.Value;
+
+/**
+ * The index progressor is a cursor like class, which allows controlled progression through the entries of an index.
+ * In contrast to a cursor, the progressor does not hold value state, but rather attempts to write the next entry to a
+ * Client. The client can them accept the entry, in which case next() returns, or reject it, in which case the
```

Comment:
Client. The client can ~them~ then accept the entry, ~~in which case next() returns~~,  or reject it, in which case the